pkg/gather/clusterconfig/operators.go

```diff
@@ -104,16 +103,69 @@ func gatherClusterOperators(ctx context.Context, configClient configv1client.Con
 	if err != nil {
 		return nil, []error{err}
 	}
+
+	// collect the cluster operators reports
+	records := clusterOperatorsReports(ctx, config.Items, dynamicClient, discoveryClient)
+
+	// check for unhealth pods
+	unhealthyPods, coReports, totalContainers := checkClusterOperator(ctx, config.Items, coreClient)
+	records = append(records, coReports...)
+	// Exit early if no unhealthy pods found
+	if len(unhealthyPods) == 0 {
+		return records, nil
+	}
+
+	// Fetch a list of containers in unhealthy pods and calculate a log size quota
+	// Total log size must not exceed maxLogsSize multiplied by logCompressionRatio
+	klog.V(2).Infof("Found %d unhealthy pods", len(unhealthyPods))
+	bufferSize := int64(recorder.MaxLogSize * logCompressionRatio / totalContainers / 2)
+	klog.V(2).Infof("Maximum buffer size: %v bytes", bufferSize)
+	buf := bytes.NewBuffer(make([]byte, 0, bufferSize))
+
+	// Fetch previous and current container logs
+	for _, isPrevious := range []bool{true, false} {
+		for _, pod := range unhealthyPods {
+			allContainers := pod.Spec.InitContainers
+			allContainers = append(allContainers, pod.Spec.Containers...)
+			for _, c := range allContainers {
+				logName := fmt.Sprintf("%s_current.log", c.Name)
+				if isPrevious {
+					logName = fmt.Sprintf("%s_previous.log", c.Name)
+				}
```

Comment:
I think we are interested in the previous log only when the pod/container was unhealthy (restarted). So I think we should keep the original check. I think the logic should be following:
- if the CO is degraded then try to collect all the related current logs
- if the CO is degraded and you found some restarted pods then try to collect the previous logs

WDYT @vrutkovs  pls?

Reply:
@rluders It looks better now, however the logic is still not correct IMO. Now it tries to read previous log for every degraded operator pod (if I am not mistaken). I don't think this is necessary. I think clusteroperator can be degraded even without any pod restart (is this assumption correct @vrutkovs pls?).

I still think we should check if the given pod is unhealthy and try to collect the previous log only in this case. And what about some test?

Reply:
@tremes check the last changes. It should be fine, I guess. If this is the expected behavior let me know, then I'll cover it with tests. BTW, I already separate the code in some way that we can implement some gorotines later one.